onnxruntime/python/onnxruntime_pybind_ortvalue.cc

```diff
@@ -166,6 +177,13 @@ void addOrtValueMethods(pybind11::module& m) {
             throw std::runtime_error("The provided device id doesn't match any available GPUs on the machine.");
           }
           allocator = GetCudaAllocator(device.Id());
+#elif USE_MIGRAPHX
+          allocator = GetMIGraphXAllocator(device.Id());
+#elif USE_ROCM
+          if (!IsRocmDeviceIdValid(logging::LoggingManager::DefaultLogger(), device.Id())) {
+            throw std::runtime_error("The provided device id doesn't match any available GPUs on the machine.");
+          }
+          allocator = GetRocmAllocator(device.Id());
```

Comment:
We might be in an odd situation here as our offering has both MIGraphx and ROCm EPs include, thus we should we get both allocators? Did you test this when we build both MIGraphX and ROCm EPs? How does the allocator work for that?

Reply:
I tested it with both included USE_ROCM and USE_MIGRAPHX and it's working well. In that case I think it's the same which allocator is used because both of them are for GPU. If you think it can be written better, I can change it or put additional condition for both of them to use specified ROCm or MGX EP.

Reply:
I tested it with both included USE_ROCM and USE_MIGRAPHX and it's working well. In that case I think it's the same which allocator is used because both of them are for GPU. If you think it can be written better, I can change it or put additional condition for both of them to use specified ROCm or MGX EP.